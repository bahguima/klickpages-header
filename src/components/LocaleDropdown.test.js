import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import LocaleDropwdown from './LocaleDropdown.vue';

describe('components/LocaleDropdown.vue', () => {
  describe('when component is mounted', () => {
    const localVue = createLocalVue();
    localVue.use(Vuex);

    const topBar = {
      namespaced: true,
    };

    let wrapper;

    describe('when is on pt-BR locale', () => {
      beforeAll(() => {
        topBar.getters = {
          config: () => ({
            locale: {
              current_locale: 'pt-BR',
              portuguese: 'Português',
              english: 'Inglês',
              spanish: 'Espanhol',
              selected: 'Selecionado',
            },
          }),
        };

        const store = new Vuex.Store({ modules: { topBar } });
        wrapper = shallowMount(LocaleDropwdown, { localVue, store });
      });

      afterAll(() => {
        wrapper.destroy();
        jest.clearAllMocks();
      });

      it('should renders html correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    describe('when is on en locale', () => {
      beforeAll(() => {
        topBar.getters = {
          config: () => ({
            locale: {
              current_locale: 'en',
              portuguese: 'Portuguese',
              english: 'English',
              spanish: 'Spanish',
              selected: 'Selected',
            },
          }),
        };

        const store = new Vuex.Store({ modules: { topBar } });
        wrapper = shallowMount(LocaleDropwdown, { localVue, store });
      });

      afterAll(() => {
        wrapper.destroy();
        jest.clearAllMocks();
      });

      it('should renders html correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    describe('when is on es locale', () => {
      beforeAll(() => {
        topBar.getters = {
          config: () => ({
            locale: {
              current_locale: 'es',
              portuguese: 'Portugués',
              english: 'Inglés',
              spanish: 'Español',
              selected: 'Seleccionado',
            },
          }),
        };

        const store = new Vuex.Store({ modules: { topBar } });
        wrapper = shallowMount(LocaleDropwdown, { localVue, store });
      });

      afterAll(() => {
        wrapper.destroy();
        jest.clearAllMocks();
      });

      it('should renders html correctly', () => {
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    describe('when localePath method is called', () => {
      const currentLocale = 'pt-BR';
      const newLocale = 'en';
      const baseUrl = 'https://sample.klickpages.com';

      beforeAll(() => {
        topBar.getters = {
          config: () => ({
            locale: {
              current_locale: currentLocale,
              portuguese: 'Português',
              english: 'Inglês',
              spanish: 'Espanhol',
              selected: 'Selecionado',
            },
          }),
        };

        const store = new Vuex.Store({ modules: { topBar } });
        wrapper = shallowMount(LocaleDropwdown, { localVue, store });

        global.window = Object.create(window);
        Object.defineProperty(window, 'location', {
          value: {
            href: baseUrl,
          },
          writable: true,
        });
      });

      afterAll(() => {
        wrapper.destroy();
        jest.clearAllMocks();
      });

      describe('when the current locale is not on the url', () => {
        beforeAll(() => {
          window.location.href = baseUrl;
        });
        it('should return just the new locale', () => {
          expect(wrapper.vm.localePath(newLocale)).toEqual(`/${newLocale}`);
        });
      });

      describe('when the current locale is in the url', () => {
        const extendedUrl = `${baseUrl}/${currentLocale}/something`;

        beforeAll(() => {
          window.location.href = extendedUrl;
        });

        it('should return the previous url with the locale replaced', () => {
          expect(wrapper.vm.localePath(newLocale)).toEqual(`${baseUrl}/${newLocale}/something`);
        });
      });
    });
  });
});
