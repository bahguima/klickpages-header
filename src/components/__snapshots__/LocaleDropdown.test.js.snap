// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when component is mounted when is on en locale should renders html correctly 1`] = `
<div
  class="dropdown drop"
  href="#"
>
  <div
    class="drop__text"
  />
   
  <a
    class="dropdown-toggle"
    data-toggle="dropdown"
    href="javascript:;"
  >
    <span
      class="image image--en"
    />
    
    en
  
  </a>
   
  <div
    class="drop__list dropdown-menu"
  >
    <div
      class="drop__shadow"
    >
      <a
        class="drop__item"
        href="/pt-BR"
      >
        <span
          class="image image--pt-BR"
        />
        
        Portuguese
        
        <span
          class="select"
        >
          Selected
        </span>
      </a>
       
      <a
        class="drop__item active"
        href="/en"
      >
        <span
          class="image image--en"
        />
        
        English
        
        <span
          class="select"
        >
          Selected
        </span>
      </a>
       
      <a
        class="drop__item"
        href="/es"
      >
        <span
          class="image image--es"
        />
        
        Spanish
        
        <span
          class="select"
        >
          Selected
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`when component is mounted when is on es locale should renders html correctly 1`] = `
<div
  class="dropdown drop"
  href="#"
>
  <div
    class="drop__text"
  />
   
  <a
    class="dropdown-toggle"
    data-toggle="dropdown"
    href="javascript:;"
  >
    <span
      class="image image--es"
    />
    
    es
  
  </a>
   
  <div
    class="drop__list dropdown-menu"
  >
    <div
      class="drop__shadow"
    >
      <a
        class="drop__item"
        href="/pt-BR"
      >
        <span
          class="image image--pt-BR"
        />
        
        Portugués
        
        <span
          class="select"
        >
          Seleccionado
        </span>
      </a>
       
      <a
        class="drop__item"
        href="/en"
      >
        <span
          class="image image--en"
        />
        
        Inglés
        
        <span
          class="select"
        >
          Seleccionado
        </span>
      </a>
       
      <a
        class="drop__item active"
        href="/es"
      >
        <span
          class="image image--es"
        />
        
        Español
        
        <span
          class="select"
        >
          Seleccionado
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`when component is mounted when is on pt-BR locale should renders html correctly 1`] = `
<div
  class="dropdown drop"
  href="#"
>
  <div
    class="drop__text"
  />
   
  <a
    class="dropdown-toggle"
    data-toggle="dropdown"
    href="javascript:;"
  >
    <span
      class="image image--pt-BR"
    />
    
    pt-BR
  
  </a>
   
  <div
    class="drop__list dropdown-menu"
  >
    <div
      class="drop__shadow"
    >
      <a
        class="drop__item active"
        href="/pt-BR"
      >
        <span
          class="image image--pt-BR"
        />
        
        Português
        
        <span
          class="select"
        >
          Selecionado
        </span>
      </a>
       
      <a
        class="drop__item"
        href="/en"
      >
        <span
          class="image image--en"
        />
        
        Inglês
        
        <span
          class="select"
        >
          Selecionado
        </span>
      </a>
       
      <a
        class="drop__item"
        href="/es"
      >
        <span
          class="image image--es"
        />
        
        Espanhol
        
        <span
          class="select"
        >
          Selecionado
        </span>
      </a>
    </div>
  </div>
</div>
`;
